tableview/src/main/java/com/evrencoskun/tableview/adapter/recyclerview/CellRowRecyclerViewAdapter.java

```diff
@@ -78,23 +91,25 @@ public int getItemViewType(int position) {
     @Override
     public void onViewAttachedToWindow(RecyclerView.ViewHolder holder) {
         super.onViewAttachedToWindow(holder);
-        AbstractViewHolder viewHolder = (AbstractViewHolder) holder;
-
-        SelectionState selectionState = mTableAdapter.getTableView().getSelectionHandler()
-                .getCellSelectionState(holder.getAdapterPosition(), mYPosition);
-
-        // Control to ignore selection color
-        if (!mTableAdapter.getTableView().isIgnoreSelectionColors()) {
-
-            // Change the background color of the view considering selected row/cell position.
-            if (selectionState == SelectionState.SELECTED) {
-                viewHolder.setBackgroundColor(mTableAdapter.getTableView().getSelectedColor());
-            } else {
-                viewHolder.setBackgroundColor(mTableAdapter.getTableView().getUnSelectedColor());
+        
+        if(mTableAdapter.getTableView().isSelectable()) {
+            AbstractViewHolder viewHolder = (AbstractViewHolder) holder;
+            SelectionState selectionState = mTableAdapter.getTableView().getSelectionHandler()
+                    .getSelectionStateCell(holder.getAdapterPosition(), mYPosition);
+
+            // Control to ignore selection color
+            if (!mTableAdapter.getTableView().isIgnoreSelectionColors()) {
+
+                // Change the background color of the view considering selected row/cell position.
+                if (selectionState == SelectionState.SELECTED) {
+                    viewHolder.setBackgroundColor(mTableAdapter.getTableView().getSelectedColor());
+                } else {
```

Comment:
Not sure, but shouldn't you explicitly handle the `UNSELECTED` and `SHADOWED` state here?